lib/iris/cube.py

```diff
@@ -2825,11 +2825,15 @@ def remap_dim_coord(coord_and_dim):
         self._dim_coords_and_dims = list(map(remap_dim_coord,
                                              self._dim_coords_and_dims))
 
-        def remap_aux_coord(coord_and_dims):
+        # Remaps all cube dimensional metadata that may have multiple
+        # dimensions.
+        def remap_cube_metadata(coord_and_dims):
```

Comment:
@lbdreyer Is there an opportunity here to take a step further and combine `remap_dim_coord` and `remap_cube_metadata`? e.g., some thing along the lines of:
```python
from collections import Iterable
...
def remap_cube_metadata(coord_and_dims):
    coord, dims = coord_and_dims
    if isinstance(dims, Iterable):
        dims = tuple(dim_mapping[dim] for dim in dims)
    else:
        dims = dim_mapping[dims]
    return coord, dims
```
Then deriving both `self._dim_coords_and_dims`, `self._aux_coords_and_dims` and `self._cell_measures_and_dims` all call the same helper function?